pints/_log_likelihoods.py

```diff
@@ -105,6 +106,61 @@ def __call__(self, x):
                       - np.sum(error**2, axis=0) / (2 * sigma**2))
 
 
+class StudentTLogLikelihood(pints.ProblemLogLikelihood):
+    """
+    *Extends:* :class:`ProblemLogLikelihood`
+
+    Calculates a log-likelihood assuming independent Student-t-distributed
+    noise at each time point, and adds two parameters: one representing
+    the degrees of freedom (``nu''), the other representing the
+    scale (``sigma'').
+
+    For a noise characterised by ``nu'' and ``sigma``, the log likelihood is of
+    the form:
+
+    .. math::
+        \log{L(\\theta, \nu, \sigma)} =
+        N\\frac{\nu}{2}\log(\nu) - N\log(\sigma) -
+        N\log B(\nu/2, 1/2)
+        -\\frac{1+\nu}{2}\sum_{i=1}^N\log(\nu +
+        \\frac{x_i - f(\\theta)}{\sigma}^2)
+
+    where B(.,.) is a beta function.
+
+    Arguments:
+
+    ``problem``
+    A :class:`SingleOutputProblem` or :class`MultiOutputProblem`. For a
+    single-output problem two parameters is added (nu, sigma), where nu is
+    the dgrees of freedom and sigma is scale, for a multi-output
+    problem ``2 * n_outputs`` parameters are added.
+    """
+    def __init__(self, problem):
+        super(StudentTLogLikelihood, self).__init__(problem)
+
+        # Get number of times, number of outputs
+        self._nt = len(self._times)
+        self._no = problem.n_outputs()
+
+        # Add parameters to problem (two for each output)
+        self._dimension = problem.n_parameters() + 2 * self._no
+
+        # Pre-calculate
+        self._N = len(self._times)
+
+    def __call__(self, x):
+        # For multiparameter problems the parameters are stored as
+        # (nu_1, sigma_1, nu_2, sigma_2,...)
+        params = x[-(2 * self._no):]
```

Comment:
Sorry, know I said it doesn't matter, but just realised it might be better to store model parameters first, distribution parameters second, so that we stay consistent with the UnknownNoiseLogLikelihood version. Sorry!

Reply:
They are currently like that. I admit my explanation was confusing though! I have added a new bit of explanation to the code to (hopefully) explain it more clearly.